src/crimson/osd/main.cc

```diff
@@ -37,8 +37,8 @@ seastar::logger& logger() {
 }
 
 void usage(const char* prog) {
-  std::cout << "usage: " << prog << " -i <ID>\n"
-            << "  --help-seastar    show Seastar help messages\n";
+  std::cout << "usage: " << prog << "\n"
+            << "  -i <ID>\n";
```

Comment:
> as we always want to print out the seastar's app options and seastar's seastar options help messages

I think this PR is to make this working.

After applying the PR, `./bin/crimson-osd --help` will print:
```
usage: ./bin/crimson-osd
  -i <ID>
  --conf/-c FILE    read configuration from the given configuration file
  --id/-i ID        set ID portion of my name
  --name/-n TYPE.ID set name
  --cluster NAME    set cluster name (default: ceph)
  --setuser USER    set uid to user or uid (and gid to user's gid)
  --setgroup GROUP  set gid to group or gid
  --version         show version and quit

  -d                run in foreground, log to stderr
  -f                run in foreground, log to usual location

  --debug_ms N      set message debug level (e.g. 1)
Crimson options:
  -h [ --help ]                       show help message
  --help-seastar                      show help message about seastar options
  --help-loggers                      print a list of logger names and exit
  --mkkey                             generate a new secret key. This is
                                      normally used in combination with --mkfs
  --mkfs                              create a [new] data directory
  --debug                             enable debug output on all loggers
  --trace                             enable trace output on all loggers
  --no-mon-config                     do not retrieve configuration from
                                      monitors on boot
  --prometheus_port arg (=0)          Prometheus port. Set to zero to disable
  --prometheus_address arg (=0.0.0.0) Prometheus listening address
  --prometheus_prefix arg (=osd)      Prometheus metrics prefix
```
And `./bin/crimson-osd --help-seastar` will unfold all the seastar helpers, `--help-loggers` will unfold all loggers. These are now handled by the seastar app itself.

Before this change, `./bin/crimson-osd --help` won't print the section of `Crimson options` because `return EXIT_SUCCESS;` prevents feeding `--help` to `app.run()`.

Reply:
i see. could you please add a short explanation in the commit message to note that, something like

> now that `app_template` handles `--help-seastar`, and prints out all app-level usages if `--help` is included in command line options. there is no need to add a dedicated `--help-seastar` by ourselves and translate it to `--help`.

Reply:
Thanks for writing down the explanation. I added it in the commit message.